pkg/provider/azure_backoff.go

```diff
@@ -272,7 +275,31 @@ func (az *Cloud) ListLB(service *v1.Service) ([]network.LoadBalancer, error) {
 		return nil, rerr.Error()
 	}
 	klog.V(2).Infof("LoadBalancerClient.List(%v) success", rgName)
-	return allLBs, nil
+
+	if !wantLb {
+		return allLBs, nil
+	}
```

Comment:
When wantLb=false, the nodes would be nil, so we cannot filter out lbs by node names. Return allLBs here because in the lb deletion reconcile loop, we don't need to filter out unmanaged lbs.